config/initializers/github_client.rb

```diff
@@ -2,12 +2,22 @@
 
 module GitHubClassroom
   def self.github_client(options = {})
-    client_options = {
-      client_id:     Rails.application.secrets.github_client_id,
-      client_secret: Rails.application.secrets.github_client_secret,
-      auto_paginate: true
-    }.merge!(options)
+    client_options = options.tap do |opts|
+      opts[:user_agent] = "GitHub Classroom"
 
+      unless options.key?(:access_token)
+        opts[:client_id]     = Rails.application.secrets.github_client_id
+        opts[:client_secret] = Rails.application.secrets.github_client_secret
+      end
+
+      if enterprise_instance?
+        opts[:api_endpoint] = "https://#{Rails.application.secrets.github_enterprise_hostname}/api/v3"
+      end
+    end
     Octokit::Client.new(client_options)
   end
+
+  def self.enterprise_instance?
+    Rails.application.secrets.enterprise_enabled == true
```

Comment:
I think this should be `github_enterprise_enabled`, additionally we should verify the data type here (boolean versus string):

```suggestion
    Rails.application.secrets.github_enterprise_enabled == true
```

Reply:
Instead, I think this might be better just detecting if a `GITHUB_ENTERPRISE_HOSTNAME` is present.